llama-index-packs/llama-index-packs-mixture-of-agents/llama_index/packs/mixture_of_agents/base.py

```diff
@@ -0,0 +1,112 @@
+# Reference: https://github.com/togethercomputer/MoA
+
+import logging
+from typing import Any, Dict, List
+import copy
+
+from llama_index.core.llama_pack.base import BaseLlamaPack
+from llama_index.core.llms.llm import LLM
+from llama_index.core.llms import ChatMessage
+
+logger = logging.getLogger(__name__)
+
+
+class MixtureOfAgentsPack(BaseLlamaPack):
+    def __init__(
+        self,
+        llm: LLM,
+        reference_llms: List[LLM],
+        num_layers: int = 3,
+    ) -> None:
+        self.llm = llm
+        self.reference_llms = reference_llms
+        self.num_layer = num_layers
+
+    def inject_references_to_messages(
```

Comment:
missing a few types and return types on a few function definitions like this one